frontend/expr_inferrer.py

```diff
@@ -375,33 +376,69 @@ def _get_args_types(args, context, instance, solver):
     return args_types
 
 
-def _infer_annotated_function_call(args_types, solver, annotations):
-    result_type = solver.new_z3_const("call")
-    solver.annotation_resolver.add_annotated_function_axioms(args_types, solver, annotations, result_type)
+def _infer_annotated_function_call(args_types, solver, annotations, result_type):
+    return solver.annotation_resolver.get_annotated_function_axioms(args_types, solver, annotations, result_type)
 
-    return result_type
+
+def _get_builtin_method_call_axioms(args_types, solver, context, result_type, method_name):
+    """Get the axioms of built-in method calls"""
+    possible_methods = context.get_matching_methods(method_name)
+    method_axioms = []
+    for method in possible_methods:
+        cur_method_axioms = _infer_annotated_function_call(args_types, solver, method, result_type)
+        if cur_method_axioms is not None:
+            method_axioms.append(cur_method_axioms)
+    return method_axioms
 
 
 def infer_func_call(node, context, solver):
```

Comment:
IMHO this needs some refactoring and some more comments because I am not really sure of what is going on. For instance, I have no idea of what the business is with `instance` that sometimes is `None` and sometimes is the result of `infer(node.func.value, context, solver)` but then can be `None` again.